In [30]:
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import datetime as dt

In [37]:
df_allmatch = pd.read_csv("./match_data_yearly/all_years.csv")
df_allmatch

ID  Year  Sec        Date                  Home  \
0      6010903  2006    1  2006-03-04           gamba-osaka   
1      6012807  2006    1  2006-03-05        ventforet-kofu   
2      6012231  2006    1  2006-03-05              fc-tokyo   
3      6011323  2006    1  2006-03-05          jubilo-iwata   
4      6010820  2006    1  2006-03-05  nagoya-grampus-eight   
...        ...   ...  ...         ...                   ...   
4585  20340120  2020   34  2020-12-19       kashima-antlers   
4586  20340810  2020   34  2020-12-19  nagoya-grampus-eight   
4587  20343331  2020   34  2020-12-19            sagan-tosu   
4588  20342218  2020   34  2020-12-19              fc-tokyo   
4589  20340314  2020   34  2020-12-19    urawa-red-diamonds   

                     Away  HomeID  AwayID  HomeGF  AwayGF  W/L Stadium  \
0      urawa-red-diamonds       9       3       1       1    0      万博   
1         shimizu-s-pulse      28       7       0       2    2      小瀬   
2            oita-trinita      22      31       2       0    1     味スタ   
3          avispa-fukuoka      13      23       1       1    0      静岡   
4            cerezo-osaka       8      20       3       2    1     瑞穂陸   
...                   ...     ...     ...     ...     ...  ...     ...   
4585         cerezo-osaka       1      20       1       1    0     カシマ   
4586  sanfrecce-hiroshima       8      10       1       0    1     豊田ス   
4587         oita-trinita      33      31       2       2    0     駅スタ   
4588          vissel-kobe      22      18       1       0    1     味スタ   
4589    consadole-sapporo       3      14       0       2    2      埼玉   

      Attendances  HomeRate  AwayRate  HomeRD  AwayRD  
0           20916      1500      1500       0       0  
1           14277      1500      1500       0       0  
2           22531      1500      1500       0       0  
3           28564      1500      1500       0       0  
4           17199      1500      1500       0       0  
...           ...       ...       ...     ...     ...  
4585        11251      1594      1560       4      -6  
4586        16825      1515      1524       9     -13  
4587         8966      1453      1458      17       7  
4588        14373      1540      1453       8       0  
4589        19319      1485      1458     -19       3  

[4590 rows x 17 columns]

In [43]:
len(df_allmatch[df_allmatch["W/L"]==0])/len(df_allmatch)

0.23485838779956428

In [44]:
df_allmatch=df_allmatch[df_allmatch["Year"]>=2013]
df_allmatch

ID  Year  Sec        Date                  Home  \
2142  13010512  2013    1  2013-03-02   yokohama-fa-marinos   
2143  13010813  2013    1  2013-03-02  nagoya-grampus-eight   
2144  13013301  2013    1  2013-03-02            sagan-tosu   
2145  13012078  2013    1  2013-03-02          cerezo-osaka   
2146  13015428  2013    1  2013-03-02        vegalta-sendai   
...        ...   ...  ...         ...                   ...   
4585  20340120  2020   34  2020-12-19       kashima-antlers   
4586  20340810  2020   34  2020-12-19  nagoya-grampus-eight   
4587  20343331  2020   34  2020-12-19            sagan-tosu   
4588  20342218  2020   34  2020-12-19              fc-tokyo   
4589  20340314  2020   34  2020-12-19    urawa-red-diamonds   

                     Away  HomeID  AwayID  HomeGF  AwayGF  W/L Stadium  \
2142      shonan-bellmare       5      12       4       2    1     日産ス   
2143         jubilo-iwata       8      13       1       1    0     豊田ス   
2144      kashima-antlers      33       1       1       1    0    ベアスタ   
2145      albirex-niigata      20      78       1       0    1      長居   
2146       ventforet-kofu      54      28       1       1    0    ユアスタ   
...                   ...     ...     ...     ...     ...  ...     ...   
4585         cerezo-osaka       1      20       1       1    0     カシマ   
4586  sanfrecce-hiroshima       8      10       1       0    1     豊田ス   
4587         oita-trinita      33      31       2       2    0     駅スタ   
4588          vissel-kobe      22      18       1       0    1     味スタ   
4589    consadole-sapporo       3      14       0       2    2      埼玉   

      Attendances  HomeRate  AwayRate  HomeRD  AwayRD  
2142        24298      1550      1312       0       0  
2143        21748      1578      1481       0       0  
2144        12728      1470      1537       0       0  
2145        15051      1490      1477       0       0  
2146        16353      1538      1398       0       0  
...           ...       ...       ...     ...     ...  
4585        11251      1594      1560       4      -6  
4586        16825      1515      1524       9     -13  
4587         8966      1453      1458      17       7  
4588        14373      1540      1453       8       0  
4589        19319      1485      1458     -19       3  

[2448 rows x 17 columns]

In [33]:
true_count = 0
for index,row in df_allmatch.iterrows():
    if  (row["HomeRate"]>row["AwayRate"] and row["W/L"] == 1):
        true_count += 1
            
    elif (row["AwayRate"]>row["HomeRate"] and row["W/L"] == 2 ):
        true_count += 1
        
    elif (row["HomeRate"] == row["AwayRate"] and row["W/L"] ==0):
        true_count += 1

In [34]:
true_count

1158

In [35]:
true_count / len(df_allmatch)

0.4730392156862745

In [47]:
df_allmatch_not0=df_allmatch[df_allmatch["W/L"]!=0]

In [48]:
true_count = 0
for index,row in df_allmatch_not0.iterrows():
    if  (row["HomeRate"]>row["AwayRate"] and row["W/L"] == 1):
        true_count += 1
            
    elif (row["AwayRate"]>row["HomeRate"] and row["W/L"] == 2 ):
        true_count += 1
        
    elif (row["HomeRate"] == row["AwayRate"] and row["W/L"] ==0):
        true_count += 1

In [49]:
true_count / len(df_allmatch_not0)

0.6156716417910447

### 考察 

単純にイロレーティングの大小で勝敗を判断すると13-20年で47%の的中率になる｡
この手法だと全試合の20-25%ほどある引き分け試合を全く当てることは出来ないので､それも大きく響いている｡

引き分け試合を除外すると61%の的中率になる｡
